### DISCUSSION TITLE
メインセクションの実装に関する調査

### DISCUSSION THREAD

**Discussion by kagomen** [lgtm-factory/lgtm-factory#159] [Scrap]
#149 実装で必要：

- 全LGTMテーマ名を取得
- css laytout grid

**Comment by kagomen** [comment]
## LGTMテーマ名を取得する方法

- クライアントサイドで取得
  - リストを作成し、コントリビューターにテーマ名を付け加えてもらう
  - リストを静的importして、Imageコンポーネントで表示

- サーバーサイドでテーマ名を取得するAPIを作成し、クライアントサイドでfetchする
  - ただ、fsモジュールが使えない？んだっけ・・・？

他に、あるかな……

**Reply by kagomen** [reply]
@kazzyfrog 
テーマ名の取得方法って、どういう感じで実装するつもりだったとかありますか？👀

**Reply by kazzyfrog** [reply]
fsモジュールの`readdir`でファイル読み込めますよ１

その際、pageファイルの`use client`は削除する必要があります！
必要に応じて、コンポーネントファイルに記載する形がいいかと👍

**Reply by kagomen** [reply]
@kazzyfrog 

やってみたら、いけました！🤩
ありがとうございます！🙇

質問なんですが、以下の記事で書かれてた「edgeでnodeモジュールが動かない」というのは、route.tsx以外のファイルでは関係ない感じなんでしょうか……？👀
https://zenn.dev/kazzyfrog/articles/error-fs-using-nextjs

**Reply by kazzyfrog** [reply]
pagesファイルでも、`export const runtime = "edge";`の記述をしたら、nodeモジュールは動かないですよ！

そうでなければ、静的ページとしてビルド時に生成されます！
ビルドは、node.jsなので使える感じですね👍

**Reply by kagomen** [reply]
> そうでなければ、静的ページとしてビルド時に生成されます！
ビルドは、node.js

なるほど……！
pagesにデプロイしてもビルドはnode.jsなんですね……難解……！！
教えていただき、ありがとうございます！！！
